### ISSUE TITLE
Wrong results with test files

### ISSUE THREAD

**Issue by sarahet** [EpigenomeClock/MouseEpigeneticClock#2]
Hi,

I downloaded the MouseEpigeneticClock package and tried to run the test files provided in the ```bismarkFiles``` folder in order to check whether I set up everything correctly. However, although the script ran smoothly I got different results of aging in the range of 22-25 instead of 7-9. 

I followed all instructions for installing dependencies, however since I use a linux machine I was not able to use ```installr```, but ```pandoc``` is installed anyways. Did you ever encounter a problem with different OS (I could not find a note concerning appropriate OS but might have missed it). Further I checked whether the input files are corrupted but this seems fine and multiple downloads did result in the same problem.

Could you provide me with an advice or idea what could have gone wrong? I'd be very grateful.

**Comment by Bonder-MJ** [comment]
Hi,

Thanks for the interest in our prediction model. To be able to help you
could you please provide me with more details on the system(s) used to test
the code plus the versions of the required packages?

Regards,
Marc

**Comment by sarahet** [comment]
Hi Marc,

I use:
* Ubuntu 16 
* R version 3.5.1
* preprocessCore version 1.42.0

Also tested R version 3.3.2 as you stated in the first issue with preprocessCore version 1.36.0 and got the same wrong result.

Thanks,
Sara

**Comment by sarahet** [comment]
Additionally, I get the same output from the script which does not require markdown, so I guess it is not an issue concerning ```rmarkdown```, ```knitr``` or ```installr```.

EDIT: Some first debugging gave me at least the same summary statistics you placed into the control html file when running the first steps of data preprocessing myself in R.

**Comment by sarahet** [comment]
If you could provide me with some intermediate results after the quantile normalization, I could also start debugging and determining the source of the problem. This way we could maybe find a solution together?

**Comment by Bonder-MJ** [comment]
Dear Sarahet,

Thanks for giving more details on the problem. I will get back to you early
next week. I will do some testing myself and let you know how to solve this.

Regards,
Marc

**Comment by sarahet** [comment]
Thanks a lot! Let me know if I can help/assist further.

**Comment by sarahet** [comment]
Dear Marc,

Together with our IT department we found the source of the problem I encountered. It is rather independent of the operating system but due to different setting in the locale LC_COLLATE which influences the underlying ```sort``` function that is used when calling ```order``` in the following line:
```filesToProcess[[i]] <- filesToProcess[[i]][order(rownames(filesToProcess[[i]])),]``` 
(line 52 in the markdown-free script)

According to the row names of your ```betas``` object that is used subsequently you sorted the coordinates character-based leading to an order like this:

```
1:109034542
1:153890242
1:158251341
1:16657135
1:180350659
1:73924900
```

This order needs to be preserved also in the input data as far as I understand which you accomplish with the ```order``` function in the line mentioned above. My local setting ```LC_COLLATE=C``` seems to be different than the one you used since it uses the underlying C implementation of sort leading to a sorting order like this (where ```0``` is valued lower than ```:```):

```
10:127326488
10:127326514
10:127326525
10:127326529
10:127326603
10:44528354
```

Therefore, the wrong CpGs are multiplied by the pre-calculated beta values. In order to avoid this issue independent of the local settings, this solution worked for me:

```filesToProcess[[i]] <- filesToProcess[[i]][rownames(betas),]```

Kind regards,
Sara
